common/config/rush/pnpm-lock.yaml

```diff
@@ -52,31 +52,31 @@ importers:
       ws: ^7.5.3
     dependencies:
       '@bentley/imodeljs-native': 4.0.2
-      '@itwin/cloud-agnostic-core': 1.5.0
+      '@itwin/cloud-agnostic-core': 1.6.0
       '@itwin/core-telemetry': link:../telemetry
-      '@itwin/object-storage-azure': 1.5.0
-      '@itwin/object-storage-core': 1.5.0
+      '@itwin/object-storage-azure': 1.6.0
+      '@itwin/object-storage-core': 1.6.0
       form-data: 2.5.1
       fs-extra: 8.1.0
       inversify: 5.0.5
       json5: 2.2.3
       multiparty: 4.2.3
-      semver: 7.3.8
+      semver: 7.5.0
       ws: 7.5.9
     devDependencies:
       '@itwin/build-tools': link:../../tools/build
       '@itwin/core-bentley': link:../bentley
       '@itwin/core-common': link:../common
       '@itwin/core-geometry': link:../geometry
       '@itwin/core-webpack-tools': link:../../tools/webpack-core
-      '@itwin/eslint-plugin': 4.0.0-dev.33_eslint@8.36.0+typescript@5.0.2
+      '@itwin/eslint-plugin': 4.0.0-dev.35_voubu7prgxjfsfbgx5d4sqnwiy
```

Comment:
Do you want to pin this to dev.33 and deal with the peer deps and new recommended config separately, or lump it all in with this PR?
FYI `auto-install-peers` in .npmrc will install the peer deps into common/temp/node_modules but `npm run lint` will try and fail to find them in the specific packages' own node_modules directories.

Reply:
im running into issues locally even after not changing the npmrc to auto-install-peers and pinning down to dev 33
<img width="1273" alt="image" src="https://user-images.githubusercontent.com/11051042/233162348-c493d274-9ea0-4d83-b2bc-10a781f3e436.png">

@paulius-valiunas i'd like for us to rethink the stance made in your eslint-plugin pr; looks like the assumption pkg managers installing peers for you doesnt work well in a monorepo

Reply:
by the way, seems like pnpm 6.32 also supports this, so we didn't have to update rush just for this. It's still a good idea to keep it up to date, though.
Source: https://twitter.com/pnpmjs/status/1528300665063321600?lang=en